util/crc32c.cc

```diff
@@ -1279,5 +1284,164 @@ uint32_t Extend(uint32_t crc, const char* buf, size_t size) {
 }
 
 
+// The code for crc32c combine, copied with permission from folly
+
+// Standard galois-field multiply.  The only modification is that a,
+// b, m, and p are all bit-reflected.
+//
+// https://en.wikipedia.org/wiki/Finite_field_arithmetic
+static constexpr uint32_t gf_multiply_sw_1(
+    size_t i, uint32_t p, uint32_t a, uint32_t b, uint32_t m) {
+  // clang-format off
+  return i == 32 ? p : gf_multiply_sw_1(
+      /* i = */ i + 1,
+      /* p = */ static_cast<uint32_t>(p ^ (static_cast<int64_t>(-((b >> 31) & 1) & a))),
```

Comment:
I guess we have to polish this line from folly to make Microsoft happy. I'm not sure what the casts are supposed to be for. It looks like it computes `/* p = */ p ^ ((0U - ((b >> 31) & 1U)) & a)`